### DISCUSSION TITLE
Example to write text to a graphic file fails on raspberry pi, Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by RajaSrinivasan** [ImageMagick/ImageMagick#4188] [Development]
### Prerequisites

- [X ] I have written a descriptive issue title
- [X ] I have verified that I am using the latest version of ImageMagick
- [ X] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Steps to Reproduce

void GraphicMarkdown::emitLine(std::string _line) {
    Magick::Image my_image( Magick::Geometry(320,220), Magick::Color("white"));
    std::vector<Magick::Drawable> text_draw_list;
    text_draw_list.push_back(
    Magick::DrawableFont("-misc-fixed-medium-o-semicondensed—13-*-*-*-c-60-iso8859-1"));
// set the text to be drawn at specified position: x=101, y=50 this case
text_draw_list.push_back( Magick::DrawableText(101, 50, "text to write on the canvas"));
// set the text color (the fill color must be set to transparent)
text_draw_list.push_back( Magick::DrawableStrokeColor(Magick::Color("black")));
text_draw_list.push_back( Magick::DrawableFillColor(Magick::Color("white")));
// draw the"text to write on the canvas" string on the canvas with the above settings*/
my_image.draw(text_draw_list) ;
}
The above is a simple copy of the example. The above when built fails with 
terminate called after throwing an instance of 'Magick::ErrorDraw'
  what():  Magick: non-conforming drawing primitive definition `text' @ error/draw.c/RenderMVGContent/4479
make: *** [Makefile:17: test] Aborted
### System Configuration

One of the systems:
Linux LAPTOP-1M5AFEIQ 4.19.104-microsoft-standard #1 SMP Wed Feb 19 06:37:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Same failure on raspberry pi - 
- ImageMagick version:
- Environment (Operating system, version and so on):

- Additional information:

Version: ImageMagick 7.0.10-30 Q16 x86_64 2020-09-15 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): fontconfig freetype pangocairo png xml zlib

**Comment by urban-warrior** [comment]
Unfortunately, we cannot reproduce the problem. The program completes without complaint. We're using

```
$ convert --version
Version: ImageMagick 7.0.10-30 Q16 x86_64 2020-09-15 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype gslib heic jbig jng jp2 jpeg lcms lzma openexr pangocairo png ps raqm raw tiff webp wmf x xml zlib
$ c++ `Magick++-config --cxxflags --cppflags` -O2 -o demo demo.cpp `Magick++-config --ldflags --libs`
$ ./demo
$
```

**Comment by RajaSrinivasan** [comment]
I built imagemagick without X11. (./configure --wihout-x) Will this make a difference?

Or I am using stdc++17 for my app.

My gcc is g++ (Ubuntu 9.3.0-10ubuntu2) 9.3.0. Can this explain.

If I understood what 

 what():  Magick: non-conforming drawing primitive definition `text' @ error/draw.c/RenderMVGContent/4479

means I will be able to troubleshoot. What is he nonconforming definition? Thanks

**Comment by fmw42** [comment]
It means your draw command is malformed and so the drawing primitive definition does not conform to the specifications.

**Comment by RajaSrinivasan** [comment]
well. This is what I am puzzled about. The code is a cut and paste from the example.#include <Magick++.h>
int main(int argc, char **argv) {

    Magick::Image my_image( Magick::Geometry(320,220), Magick::Color("cyan"));
    std::vector<Magick::Drawable> text_draw_list;
    text_draw_list.push_back(
    Magick::DrawableFont("-misc-fixed-medium-o-semicondensed—13-*-*-*-c-60-iso8859-1"));
    // set the text to be drawn at specified position: x=101, y=50 this case
    // set the text color (the fill color must be set to transparent)
    text_draw_list.push_back( Magick::DrawableStrokeColor(Magick::Color("black")));
    text_draw_list.push_back( Magick::DrawableFillColor(Magick::Color("white"))); 
    text_draw_list.push_back( Magick::DrawableText(101, 50, "text "));
    // draw the"text to write on the canvas" string on the canvas with the above settings*/
    my_image.draw(text_draw_list) ;

}

only changes are what I have been playing with. what is the "malformed" for the text? Is there an issue with the DrawableText call? Is it the font perhaps?

thanks

**Comment by fmw42** [comment]
Sorry, I do not code in that API. Some one else will have to help you further.
